tests/stepDefinitions/addPromotionContext.js

```diff
@@ -0,0 +1,28 @@
+const { Given, When, Then } = require('@cucumber/cucumber')
+const { expect } = require("@playwright/test");
+
+const { LoginPage} = require("../pageObjects/LoginPage");
+const { AddPromotion } = require("../pageObjects/AddPromotion");
+const addPromotion = new AddPromotion();
+const login = new LoginPage();
+
+
+Given('user {string} has logged in to the admin panel with following credentials:', async function (user, credentials) {
+    await login.navigateToAdminLoginPage();
+    for (const { email, password } of credentials.hashes()) {
```

Comment:
Remove the loop here. You don't need it.
```suggestion
    for (const { email, password } of credentials.hashes()) {
```